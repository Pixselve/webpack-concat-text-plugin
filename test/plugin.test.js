import fs from "fs";
import path from "path";

jest.unmock("concat");
jest.unmock("glob");

/* eslint-disable */
import { compile, createCompiler, cleanErrorStack } from "./utils";
import ConcatTextPlugin, { PLUGIN_NAME } from "../index";
/* eslint-enable */

const cases = process.env.CASES ? process.env.CASES.split(",") : fs.readdirSync(path.join(__dirname, "cases"));

describe(PLUGIN_NAME, () => {
	cases.forEach((testCase) => {
		it(testCase, (done) => {

			const testDirectory = path.join("test", "cases", testCase);
			const outputPath = path.resolve("test", "__output__", testCase);
			const filename = `${testCase}.js`;
			const configFile = path.resolve(path.join(testDirectory, "config.js"));

			const defaultConfig = {
				name: testCase,
				outputPath
			};

			// eslint-disable-next-line global-require
			const caseConfig = (fs.existsSync(configFile))
				? require(configFile)
				: {};

			const pluginConfig = Object.assign({}, defaultConfig, caseConfig);
			const webpackConfig = {
				output: {
					filename,
					path: path.join(__dirname, "__output__")
				}
			};

			const compiler = createCompiler(webpackConfig);

			new ConcatTextPlugin(pluginConfig).apply(compiler);

			compile(compiler).then((stats) => {
				const errors = stats.compilation.errors.map(cleanErrorStack);
				const warnings = stats.compilation.warnings.map(cleanErrorStack);
				const files = stats.toJson().assets.filter((asset) => asset.name !== filename);

				expect(errors).toMatchSnapshot("errors");
				expect(warnings).toMatchSnapshot("warnings");

				files.forEach((asset) => {
					if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, asset.name)) {
						expect(stats.compilation.assets[asset.name].emitted).toBeTruthy();
						expect(stats.compilation.assets[asset.name].existsAt).toEqual(path.resolve(outputPath, pluginConfig.name));
						expect(stats.compilation.assets[asset.name].source()).toMatchSnapshot("source");
					}
				});

				done();
			});
		});
	});
});
