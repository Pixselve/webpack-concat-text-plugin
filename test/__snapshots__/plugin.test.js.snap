// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConcatTextPlugin empty: errors 1`] = `Array []`;

exports[`ConcatTextPlugin empty: source 1`] = `
"
"
`;

exports[`ConcatTextPlugin empty: warnings 1`] = `Array []`;

exports[`ConcatTextPlugin multiple-files: errors 1`] = `Array []`;

exports[`ConcatTextPlugin multiple-files: source 1`] = `
"TestProperty.Value=Hello World

TestProperty2.Value=Hallo Welt

TestProperty3.One.Value=Hallo Welt
TestProperty3.Two.Value=Foo
"
`;

exports[`ConcatTextPlugin multiple-files: warnings 1`] = `Array []`;

exports[`ConcatTextPlugin nested-files: errors 1`] = `Array []`;

exports[`ConcatTextPlugin nested-files: source 1`] = `
"TestProperty.Value=Hello World

TestProperty3.One.Value=Hallo Welt
TestProperty3.Two.Value=Foo

TestProperty2.Value=Hallo Welt
"
`;

exports[`ConcatTextPlugin nested-files: warnings 1`] = `Array []`;

exports[`ConcatTextPlugin no-trailing-line-end: errors 1`] = `Array []`;

exports[`ConcatTextPlugin no-trailing-line-end: source 1`] = `
"TestProperty.Value=Hello World
TestProperty2.Value=Hallo Welt"
`;

exports[`ConcatTextPlugin no-trailing-line-end: warnings 1`] = `Array []`;
